internal/migrationhierarchy/mutator.go

```diff
@@ -39,6 +39,9 @@ func (m *MigrationHierarchyMutator) Handle(ctx context.Context, req admission.Re
 
 // UpdateRequester adds a requester annotation to the object.
 func (m *MigrationHierarchyMutator) UpdateRequester(migrationHierarchyObject danav1.MigrationHierarchy, requester string) ([]byte, error) {
+	if len(migrationHierarchyObject.Annotations) == 0 {
+		migrationHierarchyObject.Annotations = make(map[string]string)
```

Comment:
I think this is if is not reasonable, the if will work even after the changes maybe it works but annotations do not exists at all you creating an empty map. so the if should be not null.